hollow/src/main/java/com/netflix/hollow/api/producer/AbstractHollowProducer.java

```diff
@@ -837,7 +837,7 @@ private void announce(ProducerListeners listeners, HollowProducer.ReadState read
                         // have been running concurrently as primary and that would break the delta chain.
                         log.log(Level.INFO,
                                 "Fail the announcement because current producer is not primary (aka leader)");
-                        throw new IllegalStateException("Announcement failed primary (aka leader) check");
+                        throw new HollowProducer.NotAPrimaryProducerMidCycleException("Announcement failed primary (aka leader) check");
```

Comment:
nit: since it's scoped to HollowProducer, we could make the name a bit shorter-
`HollowProducer.NotPrimaryMidCycleException`
or
`HollowProducer.LostPrimaryStatusMidCycleException`

Reply:
changed it to HollowProducer.NotPrimaryMidCycleException.